### PR TITLE
build: use Java and JavaFX 21.0.1

### LINKED ISSUE #662: Upgrade to JavaFX 21.0.1
Scene Builders needs to be released with JavaFX 21.0.1 which has the fix for [Application window not activated in macOS 14 Sonoma](https://bugs.openjdk.org/browse/JDK-8315657)

### Expected Behavior

Scene Builder should work on MacOS Sonoma

### Current Behavior

Application window is not activated on MacOS Sonoma

### Steps to Reproduce

Run SB 21 on MacOS 14.x

### PR DESCRIPTION
**Pull request by abhinayagarwal**
### Issue

Fixes #662

### Progress

- [x] Change must not contain extraneous whitespace
- [x] License header year is updated, if required
- [x] The PR name must follow the [pre-defined format](https://github.com/gluonhq/scenebuilder/blob/master/CONTRIBUTING.md)
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

File: .github/workflows/bundles-kit.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
         required: true
         type: string
       java-version:
-        default: '21'
+        default: '21.0.1'
```
Comment (jperedadnr) [inline .github/workflows/bundles-kit.yml:10]: Don't you need to update as well the build, early-access and release workflows?

Comment (jperedadnr) [reply .github/workflows/bundles-kit.yml:10]: see: `error: release version 21.0.1 not supported` in the GitHub actions
Also `build.yml` uses jdk 21?

File: pom.xml [hunk 2]
```diff
@@ -16,11 +16,11 @@
     </modules>
 
     <properties>
-        <java.version>21</java.version>
-        <maven.compiler.source>${java.version}</maven.compiler.source>
-        <maven.compiler.target>${java.version}</maven.compiler.target>
-        <maven.compiler.release>${java.version}</maven.compiler.release>
-        <javafx.version>21</javafx.version>
+        <java.version>21.0.1</java.version>
```
Comment (jperedadnr) [inline pom.xml:19]: roll back this change, but rename to `java.major.version`?

Comment (abhinayagarwal) [reply pom.xml:19]: `java.version` is used in scene builder to show the java information in about dialog

Comment (jperedadnr) [reply pom.xml:19]: ok

**Review by jperedadnr — DISMISSED** [review]

**Review by jperedadnr — APPROVED** [review]

**Pull request merged**